File: ios/SharedDefaultsModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SharedDefaultsModule, NSObject)

RCT_EXTERN_METHOD(getSharedURL:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(clearSharedURL:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/SharedDefaultsModule.swift
import Foundation
import React

@objc(SharedDefaultsModule)
class SharedDefaultsModule: NSObject {

  @objc
  func getSharedURL(_ resolve: RCTPromiseResolveBlock,
                    rejecter reject: RCTPromiseRejectBlock) {
    let sharedDefaults = UserDefaults(suiteName: "group.com.memme.share")
    let value = sharedDefaults?.string(forKey: "sharedURL")
    print("🔥 Swift에서 읽은 값:", value ?? "nil")
    resolve(value)
  }

  @objc
  func clearSharedURL(_ resolve: RCTPromiseResolveBlock,
                      rejecter reject: RCTPromiseRejectBlock) {
    let sharedDefaults = UserDefaults(suiteName: "group.com.memme.share")
    sharedDefaults?.set("", forKey: "sharedURL")
    resolve(true)
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}